File: c++/S02-conditional-control-structure/E01-palindrome-check.cpp
#include <iostream>

using namespace std;

int main() {
    int inputNumber, outputNumber;
	int firstDigit, secondDigit, thirdDigit, fourthDigit, fifthDigit;
     
    cout << "\n\e[0;35m[========= PALINDROME CHECK =========]\e[0m\n" << endl;

    cout << "Enter the number: ";
    cin >> inputNumber;

	fifthDigit = inputNumber / 10000;
	fourthDigit = (inputNumber % 10000) / 1000;
	thirdDigit = (inputNumber % 1000) / 100;
	secondDigit = (inputNumber % 100) / 10;
	firstDigit = inputNumber % 10;

    outputNumber = fifthDigit * 10000 +  fourthDigit * 1000 + thirdDigit * 100 + secondDigit * 10 + firstDigit;

    if (inputNumber == outputNumber) {
		printf("\nThe number %i is capicua\n", inputNumber);
    } else{
		printf("\nThe number %i is not capicua\n", inputNumber);
    }

    return 0;
}

File: python/S02-conditional-control-structure/E01-palindrome-check.py
def main():
    print("\n\033[35m[========= PALINDROME CHECK =========]\033[0m\n")

    number = int(input("Enter a number: "))

    if (number > 99999 or number < 10000):
        print("Please enter a 5 digit number.")
        return
    
    inverted = int(str(number)[::-1])

    if (number == inverted):
        print(f"\033[32m{number}\033[0m is a palindrome number.")
    else:
        print(f"\033[31m{number}\033[0m is not a palindrome number.")


if __name__ == "__main__":
    main()
